packages/transport/test/modules/relationships/TerminateDecompose.test.ts

```diff
@@ -0,0 +1,49 @@
+import { IDatabaseConnection } from "@js-soft/docdb-access-abstractions";
+import { AccountController, CoreId, RelationshipChangedEvent, Transport } from "../../../src";
+import { TestUtil } from "../../testHelpers/TestUtil";
+
+describe("Terminate and Decompose simultaneously", function () {
+    let connection: IDatabaseConnection;
+
+    let transport: Transport;
+
+    let sender: AccountController;
+    let recipient: AccountController;
+
+    let relationshipId: CoreId;
+
+    beforeAll(async function () {
+        connection = await TestUtil.createDatabaseConnection();
+        transport = TestUtil.createTransport(connection);
+
+        await transport.init();
+
+        const accounts = await TestUtil.provideAccounts(transport, 2);
+        sender = accounts[0];
+        recipient = accounts[1];
+
+        const relationships = await TestUtil.addRelationship(sender, recipient);
+        relationshipId = relationships.acceptedRelationshipFromSelf.id;
+    });
+
+    afterAll(async function () {
+        await sender.close();
+        await recipient.close();
+        await connection.close();
+    });
+
+    test("should only get one RelationshipChangedEvent for Terminate and Decompose at the same time", async function () {
+        await sender.relationships.terminate(relationshipId);
+        await sender.relationships.decompose(relationshipId);
+
+        const events: RelationshipChangedEvent[] = [];
+        transport.eventBus.subscribe(RelationshipChangedEvent, function (event) {
+            events.push(event);
+        });
+
+        const relationships = await TestUtil.syncUntilHasRelationships(recipient);
+        expect(relationships).toHaveLength(1);
+
```

Comment:
Why is this here?

Reply:
right, this should not be there :)

Reply:
However, why did the test fail here before and returned 2 relationships? 👀

Reply:
it did? I don't care.

Reply:
ah, 2 events so 2 "changed" relationships

Reply:
also a "bug" but not part of this PR.